Bugzilla/Template.pm

```diff
@@ -270,12 +272,14 @@ sub renderComment {
         $text =~ s/\x{FDD2}($i)\x{FDD3}/$things[$i]/eg;
     }
 
-    if ($skip_markdown) {
-        return $text;
-    }
-    else {
-        return Bugzilla->markdown_parser->render_html($text);
+    unless ($skip_markdown) {
+        $text = decode('UTF-8', Bugzilla->markdown_parser->render_html($text));
     }
+
+    my $dom = Mojo::DOM->new($text);
+    my $urlbase = Bugzilla->localconfig->{urlbase};
+    $dom->find("a[href]:not([href^=$urlbase])")->map(attr => rel => 'nofollow');
```

Comment:
Well, Unicode characters and external links are fine, but looks like I've found an issue: internal bug links like `Bug 123` also have `rel="nofollow"`.

Reply:
Also, not related to this PR, but I see a duplicated, empty `<a>` tag for the first bug link in a rendered comment. For example,

```
http://bmo-web.vm/show_bug.cgi?id=1 or Bug 1, Bug 2
```

turns into:

```html
<a class="bz_bug_link 
bz_status_NEW " title="NEW - First Love" href="show_bug.cgi?id=1"></a><a href="http://bmo-web.vm/show_bug.cgi?id=1">http://bmo-web.vm/show_bug.cgi?id=1</a> or <a class="bz_bug_link 
bz_status_NEW " title="NEW - First Love" href="show_bug.cgi?id=1">Bug 1</a>, <a class="bz_bug_link 
bz_status_NEW " title="NEW - Yo" href="show_bug.cgi?id=2">Bug 2</a>
```

🤔

Reply:
this has been tricky but I finally have a fix. will post in the morning

Reply:
what sort of performance hit does running each comment through a DOM parser have?